kolibri/content/api.py

```diff
@@ -0,0 +1,222 @@
+
+@can_get_content_with_id
+def children_of_kind(channel_id=None, content=None, kind=None, **kwargs):
+    """The content can be a ContentMetadatas object or a content_id string."""
+    """return all ContentMetadatas of a particular kind under the given ContentMetadata"""
+    return content.get_descendants(include_self=False).filter(kind=kind).using(channel_id)
+
+# def scan_and_update_file_availability(channel_id=None):
+#     """this is a safty check to ensure that the File model truthfully reflects what's actually in and not in the content_files folder"""
+#     all_files = KolibriContent.File.objects.using(channel_id).all().only("available", 'checksum', 'extension')
+#     for f in all_files:
+#         the_file_path = settings.CONTENT_COPY_DIR + '/' + f.checksum + '.' + f.extension
+#         real_available = os.path.exists(the_file_path)
+#         if f.available !=  real_available:
+#             f.available = real_available
+
+#     bulk_update(all_files, update_fields=['available'], using=channel_id)
+
+def update_content_copy(file_object=None, content_copy=None):
+    """you can pass None on content_copy to remove the associated file on disk"""
+    if not file_object:
+        raise TypeError( "must provide a File object to update content copy")
+    if content_copy:
+        file_object.content_copy = DjFile(file(content_copy))
+        file_object.save()
+    else:
+        file_object.content_copy = None
+
+    file_object.save()
+
+
+"""channel API methods"""
+
+def process_channel_identifier(channel_identifier):
+    """this func will take a channel id or channel name and return a ChannelMetadata object"""
+    if is_valid_uuid(channel_identifier):
+        try:
+            the_channel = KolibriContent.ChannelMetadata.objects.get(channel_id=channel_identifier)
+        except KolibriContent.ChannelMetadata.DoesNotExist:
+            # no employee found
+            raise Exception("cannot find any channel with this channel id")
+        return the_channel
+    elif isinstance(channel_identifier, str):
+        try:
+            the_channel = KolibriContent.ChannelMetadata.objects.get(name=channel_identifier)
+        except KolibriContent.ChannelMetadata.DoesNotExist:
+            # no employee found
+            raise Exception("cannot find any channel with this channel name")
+        except KolibriContent.ChannelMetadata.MultipleObjectsReturned:
+            # what to do if multiple employees have been returned?
+            raise Exception("more than one channel named " + "'" + channel_identifier + "'")
+        return the_channel
+    else:
+        raise ValueError('the channel_identifier must be either a UUID String or a normal String')
+
+def get_available_channels():
+    return KolibriContent.ChannelMetadata.objects.all()
+
+def get_channel(channel_identifier):
+    return process_channel_identifier(channel_identifier)
```

Comment:
This pattern is repeated a lot... maybe you should write a `get_channel_property(channel_identifier, property_name)` and have it raise `KeyError` if it's an invalid property?